### PR TITLE
oelint-adv: init at 3.25.0, pythonPackages.oelint-parser: init at 2.11.3

### PR DESCRIPTION
**Pull request by otavio**
## Description of changes

Initializes `oelint-adv` at version 3.25.0 and its dependency `pythonPackages.oelint-parser` at version 2.11.3

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/oe/oelint-adv/Revert-urllib3-bump.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
++++ b/requirements.txt
+@@ -1,5 +1,5 @@
+ anytree ~= 2.9
+ colorama ~= 0.4
+ oelint-parser ~= 2.11
+-urllib3 ~= 2.0
+-wheel ~= 0.41
+\ No newline at end of file
++urllib3
++wheel ~= 0.41
```
Comment (thiagokokada) [inline pkgs/by-name/oe/oelint-adv/Revert-urllib3-bump.patch:23]: Can you please use `pythonRelaxDepsHook` instead?

File: pkgs/by-name/oe/oelint-adv/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    anytree
+    colorama
+    oelint-parser
+    urllib3
+  ];
+
+  patches = [
+    # Allow use with urllib3 1.26
+    ./Revert-urllib3-bump.patch
+  ];
```
Comment (thiagokokada) [inline pkgs/by-name/oe/oelint-adv/package.nix:32]:
```suggestion
  nativeBuildInputs = with python3.pkgs; [
    pip
    pythonRelaxDepsHook
  ];

  propagatedBuildInputs = with python3.pkgs; [
    anytree
    colorama
    oelint-parser
    urllib3
  ];

  pythonRelaxDeps = [
    "urllib3"
  ];
```

File: pkgs/by-name/oe/oelint-adv/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  src = fetchPypi {
+    inherit version;
+    pname = "oelint_adv";
+    hash = "sha256-dhTS2DZ7Usb1jgBv9Wm86w8CCMt64aHyBrxucLZUQjs=";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [
+    pip
+  ];
```
Comment (thiagokokada) [inline pkgs/by-name/oe/oelint-adv/package.nix:20]: Are you sure this is necessary?

File: pkgs/by-name/oe/oelint-adv/Revert-urllib3-bump.patch (hunk 1 above)
Comment (otavio) [reply pkgs/by-name/oe/oelint-adv/Revert-urllib3-bump.patch:23]: I tried but it fails as:

```
error: builder for '/nix/store/58wldb348bikm2f7kv3b6n1sfmyq75rb-oelint-adv-3.25.0.drv' failed with exit code 1;
       last 10 log lines:
       >   ir_d = dist.fetch_build_eggs(dist.install_requires)
       > WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
       > WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff580e620>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/urllib3/
       > WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff580e410>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/urllib3/
       > WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff580f460>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/urllib3/
       > WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff580f610>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/urllib3/
       > WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff580f7c0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/urllib3/
       > ERROR: Could not find a version that satisfies the requirement urllib3~=2.0 (from versions: none)
       > ERROR: No matching distribution found for urllib3~=2.0
       > error: Command '['/nix/store/lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/bin/python3.10', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/build/tmp04vys396', '--quiet', 'urllib3~=2.0']' returned non-zero exit status 1.
```

File: pkgs/by-name/oe/oelint-adv/package.nix (hunk 3 above)
Comment (otavio) [reply pkgs/by-name/oe/oelint-adv/package.nix]: Fixed.

**Comment by mweinelt** [comment]
Triggering eval after it was fixed on master.

@ofborg eval

**Review by thiagokokada — APPROVED** [review]

File: pkgs/by-name/oe/oelint-adv/package.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    anytree
+    colorama
+    oelint-parser
+    urllib3
+  ];
+
+  pythonRelaxDeps = [ "urllib3" ];
+
+  # Fail to run inside the code the build.
+  doCheck = false;
```
Comment (thiagokokada) [inline pkgs/by-name/oe/oelint-adv/package.nix:28]: Can we have a `pythonImportCheck` here?

Comment (otavio) [reply pkgs/by-name/oe/oelint-adv/package.nix:29]: Fixed.

File: pkgs/by-name/oe/oelint-adv/Revert-urllib3-bump.patch (hunk 1 above)
Comment (otavio) [reply pkgs/by-name/oe/oelint-adv/Revert-urllib3-bump.patch]: Fixed.

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**